src/Stack/Init.hs

```diff
@@ -113,6 +117,31 @@ initProject currDir initOpts = do
     liftIO $ L.writeFile dest' $ B.toLazyByteString $ renderStackYaml p
     $logInfo $ "Wrote project config to: " <> T.pack dest'
 
+-- | See https://github.com/commercialhaskell/stack/issues/1198.
```

Comment:
I didn't realize that this would require another function.  While it's not very complicated, we should avoid extra code when possible.  I think it'd be fine to only emit the warnings for `build`.